### ISSUE TITLE
[Navigation]: Updates

### ISSUE THREAD

**Issue by fallaciousreasoning** [brave/leo#605]
@AlanBreck @aguscruiz was just trying to replace the customize dialog on the new tab page with the Navigation component. It's pretty good but there are a few things that look off / not as good as the old one:
1. The active indicator doesn't animate (this is pretty important, it looks really good in the current one)
2. The hover effects don't look quite right to me

Old:
[Screencast from 2024-03-27 11-25-46.webm](https://github.com/brave/leo/assets/7678024/c5b38b2a-8ff9-44da-8fbc-53b1b641ee52)

New:
[Screencast from 2024-03-27 11-25-02.webm](https://github.com/brave/leo/assets/7678024/cd84dc4c-14c6-4b2f-af47-34271f6b04fd)

**Comment by aguscruiz** [comment]
Yeah, agreed on both. We need the animation to make it look good.

And the hover effect, yes, it does look weird, huh. 

Would it be a lot of work to convert the layout to this? It's the layout I'm using in other places throughout the products. 

- Adds a gray background with a small white padding around it, to delineate the difference between content and navigation.
- Unifies the nav with the rest of the page

https://www.figma.com/file/lVTufBkl5eNrtOQjlAHxjb/Search-box-in-NTP?type=design&node-id=351%3A3976&mode=design&t=Hq6kkRtm6oYVTi9f-1

![image](https://github.com/brave/leo/assets/15734630/f4fe838f-1008-47e2-82d0-61a416c7cb06)


If this is too much of an effort for now, we can just add a divider line between nav and content, but it would be cool to do this so it matches the other places (Like the settings redesign I have here https://www.figma.com/file/IXNCYwXmMWaF6u7Pdv1pOW/Desktop-settings---Nala?type=design&node-id=1025%3A12750&mode=design&t=hlfcLNaOhCrhCZR2-1)


Let me know

**Comment by AlanBreck** [comment]
> And the hover effect, yes, it does look weird, huh. I think it's because it doesn't have a right edge to go against and it's just floating there.

Hmmm... any recommended solutions?

> Would it be a lot of work to convert the layout to this? It's the layout I'm using in other places throughout the products.

I think this is out of scope for the navigation Navigation component.

**Comment by fallaciousreasoning** [comment]
Mmmh, I don't really like the additional padding - what if we made the background of the content area that gray?

**Comment by aguscruiz** [comment]
OK then I propose this, remove that padding that gives it a white background, but still keep the gray as a padding. And a divider between top title and the bottom content area

I don't think we can get away with just a gray background for the content, specially on the other pages in that modal. We need the white background. 

![image](https://github.com/brave/leo/assets/15734630/506ffbd2-2593-4225-99ca-14c392a1b3c2)
